llarp/crypto/crypto.cpp

```diff
@@ -81,13 +81,13 @@ namespace llarp
 
   static bool
   dh_server_priv(
-      llarp::SharedSecret& shared, const PubKey& pk, const SecretKey& sk, const TunnelNonce& n)
+      llarp::SharedSecret& shared, const PubKey& pk, const SecretKey& sk, const SymmNonce& n)
   {
     llarp::SharedSecret dh_result;
 
     if (dh(dh_result, pk, sk.toPublic(), pk.data(), sk))
     {
-      return crypto_generichash_blake2b(shared.data(), 32, n.data(), 32, dh_result.data(), 32)
```

Comment:
General thought in regards to most of these crypto functions that jason brought up the other day...

We should be checking sizes on the objects and throwing if it's not correct

Reply:
Agreed.  For most of the code this is achieved via type safety, but there may yet be bits which call the crypto functions which take bare pointers directly.  I don't know if it's in scope for this PR to seek out all of those cases, but I can if you want.

Reply:
Yeah i agree, though maybe we shouldn't rely too hard on the type safety lol. As to the pointers, we should const the ones that correspond to const references to objects in the other overloads, ex:
```C++
void do_crypto_thangs(const SecretKey& sk, const PubKey& pk, ...);

void do_crypto_thangs(const uint8_t* sk, size_t sk_size, const uint8_t* pk, size_t pk_size, ...);
```

I'll make a jira ticket RE the things we need to do to `crypto/` generally (make the uint8_t pointers const where needed, etc), and that can be it's own standalone PR? It's probably a solid 1-2 day task, so either you or i can knock it out when it makes sense between tasks

Reply:
I think it should be a separate PR, yeah.  Sounds good.